### DISCUSSION TITLE
Dunst formatting: how to align items horizontally (tabulation)?

### DISCUSSION THREAD

**Discussion by JayBeeDe** [dunst-project/dunst#1377] [Q&A]
Hi,
I have the following configuration for the format global setting:
`"%p%n <big>%s</big>     |     %b"`

![image](https://github.com/user-attachments/assets/5369035f-326b-4130-b4be-3a5503b60629)

Is there a way to have the pipe separator always aligned at the same position ? Or to set a maximum summary character length ?
If not possible, do you have an idea of what could be implemented ? So I can create an issue...

**Comment by bynect** [comment]
I think that right now it is not possible, but it should be easy to implement as a feature

**Comment by JayBeeDe** [comment]
So I think I'll create an issue, I'm just wondering what implementation should be described in the issue...
I was thinking something simple such as:
Add tabulation `\t` character compatibility to the format setting, so we would write something like this in my case: `%p%n <big>%s</big>\t|\t%b`
or do you have a better idea ?

**Reply by bynect** [reply]
I think its better to add a maximum title length. We already have settings for ellipsizing long bodies so it should not be too hard

**Reply by JayBeeDe** [reply]
"Maximum" summary length supposes that it ellipses once x length is reached. But what if notification summary is less than x length ? It should fill with spaces
Maybe we should consider another setting name ?
What about summary_width setting ?
Values can be "auto" (default current behavior) or a number that represents a fixed width in pixels of the "%s" format (remaining space will be filled with space)

**Reply by bynect** [reply]
> "Maximum" summary length supposes that it ellipses once x length is reached. But what if notification summary is less than x length ? It should fill with spaces
> Maybe we should consider another setting name ?
> What about summary_width setting ?
> Values can be "auto" (default current behavior) or a number that represents a fixed width in pixels of the "%s" format (remaining space will be filled with space)

Maybe what you can do can be done out of the box by pango markup? Thinking about it there should be some option to pad spaces in it

**Reply by JayBeeDe** [reply]
No, not found in the pango markup: https://docs.gtk.org/Pango/pango_markup.html

**Reply by bynect** [reply]
ok pango markup has less things than what I thought. I don't know what the best solution would be. probably add a way to preprocess notification with a script or something would be a more general solution
